basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/autoregistration/AutoRegisterAASAggregator.java

```diff
@@ -0,0 +1,90 @@
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+package org.eclipse.basyx.components.aas.autoregistration;
+
+import java.util.Collection;
+
+import org.eclipse.basyx.aas.aggregator.api.IAASAggregator;
+import org.eclipse.basyx.aas.metamodel.api.IAssetAdministrationShell;
+import org.eclipse.basyx.aas.metamodel.map.AssetAdministrationShell;
+import org.eclipse.basyx.aas.metamodel.map.descriptor.AASDescriptor;
+import org.eclipse.basyx.aas.registration.api.IAASRegistry;
+import org.eclipse.basyx.submodel.metamodel.api.identifier.IIdentifier;
+import org.eclipse.basyx.vab.exception.provider.ResourceNotFoundException;
+import org.eclipse.basyx.vab.modelprovider.api.IModelProvider;
+
+/**
+ * An {@link IAASAggregator} which automatically registers created Shells in the
+ * registry
+ * 
+ * @author fried
+ *
+ */
+public class AutoRegisterAASAggregator implements IAASAggregator {
+
+	private IAASAggregator aggregator;
+	private IAASRegistry registry;
+	private String endpoint;
+
+	public AutoRegisterAASAggregator(IAASAggregator aggregator, IAASRegistry registry, String endpoint) {
+		this.aggregator = aggregator;
+		this.registry = registry;
+		this.endpoint = endpoint;
+	}
+
+	@Override
+	public Collection<IAssetAdministrationShell> getAASList() {
+		return aggregator.getAASList();
+	}
+
+	@Override
+	public IAssetAdministrationShell getAAS(IIdentifier aasId) throws ResourceNotFoundException {
+		return aggregator.getAAS(aasId);
+	}
+
+	@Override
+	public IModelProvider getAASProvider(IIdentifier aasId) throws ResourceNotFoundException {
+		return aggregator.getAASProvider(aasId);
+	}
+
+	@Override
+	public void createAAS(AssetAdministrationShell aas) {
+		registry.register(new AASDescriptor(aas, endpoint));
+		aggregator.createAAS(aas);
+
+	}
+
+	@Override
+	public void updateAAS(AssetAdministrationShell aas) throws ResourceNotFoundException {
+		aggregator.updateAAS(aas);
+	}
+
+	@Override
+	public void deleteAAS(IIdentifier aasId) {
+		registry.delete(aasId);
```

Comment:
De-Registration should happen only after the aggregator successfully deleting the AAS to avoid transient inconsistent states.